worker-message-prioritization-target-capacity-calculators/src/main/java/com/github/workerframework/workermessageprioritization/targetqueue/K8sTargetQueueSettingsProvider.java

```diff
@@ -0,0 +1,184 @@
+            {
+                @Override
+                public TargetQueueSettings load(@Nonnull final Queue queue)
+                {
+                    return getTargetQueueSettingsFromKubernetes(queue);
+                }
+            });
+
+        this.kubernetesNamespaces = kubernetesNamespaces;
+    }
+
+    @Override
+    public TargetQueueSettings get(final Queue targetQueue)
+    {
+        try {
+            return targetQueueToSettingsCache.get(targetQueue);
+        } catch (final ExecutionException executionException) {
+            LOGGER.error(String.format("Cannot get settings for the %s queue as an ExecutionException was thrown. "
+                + "Falling back to using max length of %s and eligible for refill percentage of %s",
+                                       targetQueue,
+                                       TARGET_QUEUE_MAX_LENGTH_FALLBACK,
+                                       TARGET_QUEUE_ELIGIBLE_FOR_REFILL_PERCENTAGE_FALLBACK),
+                         executionException);
+
+            return new TargetQueueSettings(TARGET_QUEUE_MAX_LENGTH_FALLBACK, TARGET_QUEUE_ELIGIBLE_FOR_REFILL_PERCENTAGE_FALLBACK);
+        }
+    }
+
+    private TargetQueueSettings getTargetQueueSettingsFromKubernetes(final Queue targetQueue)
+    {
+        // Get the target queue name
+        final String targetQueueName = targetQueue.getName();
+
+        // Loop through all provided namespaces
+        for (final String kubernetesNamespace : kubernetesNamespaces) {
+
+            try {
+                // Loop through all deployments
+                for (final V1Deployment deployment : Kubectl.get(V1Deployment.class).namespace(kubernetesNamespace).execute()) {
+
+                    // Get the metadata
+                    final V1ObjectMeta metadata = deployment.getMetadata();
+                    if (metadata == null) {
+                        continue;
+                    }
+
+                    // Get the labels from the metadata
+                    final Map<String, String> labels = metadata.getLabels();
+                    if (labels == null) {
+                        continue;
+                    }
+
+                    // Check if there is a target queue name label
+                    if (!labels.containsKey(MESSAGE_PRIORITIZATION_TARGET_QUEUE_NAME_LABEL)) {
+                        continue;
+                    }
+
+                    // Check if the target queue name label value matches the target queue name provided to this method
+                    if (!labels.get(MESSAGE_PRIORITIZATION_TARGET_QUEUE_NAME_LABEL).equals(targetQueueName)) {
+                        continue;
+                    }
+
+                    // Check if there is a target queue max length label
+                    if (!labels.containsKey(MESSAGE_PRIORITIZATION_TARGET_QUEUE_MAX_LENGTH_LABEL)) {
+                        // Throw RuntimeException as this indicates a deployment error that should never happen in production
```

Comment:
Should a deployment have an invalid label value the distrubutor will fail entirely, do we really want this?